sql/core/src/test/scala/org/apache/spark/sql/sources/v2/DataSourceV2SQLSuite.scala

```diff
@@ -1646,4 +1651,70 @@ class DataSourceV2SQLSuite extends QueryTest with SharedSQLContext with BeforeAn
       }
     }
   }
+
+  test("ShowTables using v2 catalog") {
+    spark.sql("CREATE TABLE testcat.db.table_name (id bigint, data string) USING foo")
+    spark.sql("CREATE TABLE testcat.n1.n2.db.table_name (id bigint, data string) USING foo")
+
+    {
```

Comment:
Nit: It is strange to me to use blocks. I would rather get rid of `tables` and put the `runShowTablesSql` inside `assert`.

Reply:
Fixed as suggested.